### PR TITLE
refactor(android/app): Allow SystemKeyboard to install default keyboard and dictionary

### PR DESCRIPTION
**Pull request by darcywong00**
Follows #10782 for addressing #7712

This refactors the Android app so either `MainActivity` or `SystemKeyboard` can install the default keyboard/dictionary, depending whichever gets launched first.

This way, if users never launch the Keyman app but enable it as a system keyboard, they can still use the sil_euro_latin keyboard with nrc.en.mtnt dictionary.

First commit just moves existing code from `MainActivity` to `DefaultLanguageResource`.

## User Testing
Expands on test from #7712

**Setup** - This involves modifying Android Studio configuration so the IDE will install the PR build of Keyman for Android, without launching the app.
1. From Android Studio --> Run --> Edit Configurations
2. In the Run/Debug Configurations dialog, change "Launch Options --> Launch" dropdown from "Default Activity" to "Nothing". 
edit configuration
3. **For each test** on an Android device/emulator, do a clean install of the PR build of Keyman for Android. (If Keyman was previously installed, uninstall it first)
4. When testing is complete,  revert the Android Studio Configuration settings so future PR testing with launch the app.

* **TEST_INAPP_KEYBOARD** - Verifies in-app default keyboard works
1. On the Android device, launch Keyman for Android
2. Once Keyman loads, dismiss the "Get Started Menu".
3. Verify the OSK appears along with English suggestions
4. Verify you can type with the Keyman keyboard

* **TEST_SYSTEM_KEYBOARD** - Verifies system default keyboard works
(remember setup step 3. If Keyman was previously installed for another test, uninstall and do a clean installation of Keyman)
1. On the Android device, go to the Android Settings --> Languages menu and enable Keyman as a system keyboard
2. Launch Chrome and click on the textbar
3. Select Keyman as the system keyboard
4. Once Keyman loads, verify the OSK appears along with English suggestions
5. Verify you can type with the Keyman keyboard

### CODE REVIEW

**Comment by bharanidharanj** [comment]
* **TEST_INAPP_KEYBOARD (PASSED):** Tested with the attached PR build (Keyman 17.0.272-beta-test-10794) on an Android 12 / API 31 emulator and here is my observation: 1. Uninstalled the previous keyman build. 2. Opened Debug Configurations dialog, then set Launch Options to "Nothing". 3. Installed the attached PR build. 4. Opened Keyman In-App. 5. Verified the OSK appeared with English suggestions. 6. Verified that it could be possible to type with Keyman Keyboard. 

* **TEST_SYSTEM_KEYBOARD (PASSED):** 1. Set the Keyman keyboard as the system keyboard. 2. Launched Chrome browser. Clicked the text bar. 3. Verified that the Keyman Keyboard appeared along with English suggestions. 4. Also, verified that it could be possible to type with Keyman keyboard.

File: android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java [hunk 1]
```diff
@@ -64,11 +64,13 @@ public void onCreate() {
       KMManager.setDebugMode(true);
     }
     KMManager.addKeyboardEventListener(this);
-    KMManager.initialize(getApplicationContext(), KeyboardType.KEYBOARD_TYPE_SYSTEM);
-    interpreter = new KMHardwareKeyboardInterpreter(getApplicationContext(), KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    Context context = getApplicationContext();
+    KMManager.initialize(context, KeyboardType.KEYBOARD_TYPE_SYSTEM);
+    DefaultLanguageResource.install(context);
```
Comment (mcdurdin) [inline android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java:69]: Will this reinstall sil_euro_latin if the user has deliberately removed it?

**Review by mcdurdin — APPROVED** [review]

File: android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java (hunk 1 above)
Comment (darcywong00) [reply android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java:69]: DefaultLanguageReference line 34 stores a boolean if the default dictionary has ever been installed

```
editor.putBoolean(defaultKeyboardInstalled, true);
```
So future installs are skipped.

Comment (mcdurdin) [reply android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java:69]: >  default dictionary

or default keyboard?

Comment (darcywong00) [reply android/KMAPro/kMAPro/src/main/java/com/keyman/android/SystemKeyboard.java:69]: ah yeah
l 34 stores if default keyboard has ever been installed
l 58 stores if default dictionary has ever been installed

**Pull request merged**